File: Java/RefactoringWorkshop/src/main/java/coupler/featureenvy/BenefitService.java
package coupler.featureenvy;

public class BenefitService {

    private Employee employee;

    public BenefitService(Employee employee) {
        this.employee = employee;
    }

    public String employeeDetailsXML() {
        StringBuilder details = new StringBuilder();
        details.append("<EMPLOYEE>");
        details.append("<NAME>" + employee.getName() + "</NAME>");
        details.append("<BENEFIT_TYPE>" + getBenefitType() + "</BENEFIT_TYPE");
        details.append("</EMPLOYEE>");
        return details.toString();
    }

    private Benefit getBenefitType() {
        if(employee.grade() <= 3) {
            return Benefit.HEALTHCARE_BENEFIT;
        }
        if(employee.grade() <= 5) {
            return Benefit.WELLNESS_BENEFIT;
        }
        return Benefit.RETIREMENT_BENEFIT;
    }
}

File: C#/RefactoringWorkshop/RefactoringWorkshop/Coupler/FeatureEnvy/BenefitService.cs
using System.Text;

namespace RefactoringWorkshop.Coupler.FeatureEnvy
{
    public class BenefitService
    {
        private readonly Employee _employee;

        public BenefitService(Employee employee) {
            _employee = employee;
        }

        public string EmployeeDetailsXml() {
            StringBuilder details = new StringBuilder();
            details.Append("<EMPLOYEE>");
            details.Append("<NAME>" + _employee.Name + "</NAME>");
            details.Append("<BENEFIT_TYPE>" + GetBenefitType() + "</BENEFIT_TYPE");
            details.Append("</EMPLOYEE>");
            return details.ToString();
        }

        private Benefit GetBenefitType() {
            if(_employee.Grade <= 3) {
                return Benefit.HealthcareBenefit;
            }
            if(_employee.Grade <= 5) {
                return Benefit.WellnessBenefit;
            }
            return Benefit.RetirementBenefit;
        }

    }
}